src/Controls/src/Core/BindableObject.cs

```diff
@@ -482,7 +482,7 @@ public void SetValue(BindableProperty property, object value)
 			if (property == null)
 				throw new ArgumentNullException(nameof(property));
 
-			if (value is BindingBase binding && !property.ReturnType.IsAssignableFrom(typeof(BindableProperty)))
+			if (value is BindingBase binding && !property.ReturnType.IsAssignableFrom(typeof(BindingBase)))
```

Comment:
when we set BP of type BindingBase, we do not SetBinding, we SetValue. XAML inflators are doing this right, and it only happens when used in code. this is why it was never caught